src/resource_assigner/models.py

```diff
@@ -111,3 +111,22 @@ def is_template(self):
     @property
     def resource_id_list(self):
         return list(self.resources.values_list("id", flat=True))
+
+
+class TaskRuleAssignment(BaseModel):
+    """
+    Represents the assignment rules to tasks.
+    one task can have multiple rules.
+    """
+    
+    task = models.ForeignKey(Task, on_delete=models.DO_NOTHING)
```

Comment:
This should be `CASCADE` on_delete. Same goes with the AssignmentRule field. If any of those are deleted, the TaskRuleAssignment should be deleted, as well.